docs/user/task-manager/task-manager-troubleshooting.asciidoc

```diff
@@ -0,0 +1,699 @@
+https://support.elastic.co/customers/s/login/[Elastic Support portal].
+Or post in the https://discuss.elastic.co/[Elastic forum].
+
+
+[float]
+[[task-manager-health-scheduled-tasks-small-schedule-interval-run-late]]
+=== Tasks with Small Schedule Intervals Run Late
+
+*Symptom*:
+
+Tasks are scheduled to run every 2 seconds but they seem to be running too late
+
+*Resolution*:
+
+Task Manager polls for tasks at the cadence specified by the <<task-manager-settings,`xpack.task_manager.poll_interval`>> setting, which is 3 seconds by default. This means that a Task could run late if it uses a schedule which is smaller than this setting.
+
+This can be addressed by adjusting the <<task-manager-settings,`xpack.task_manager.poll_interval`>> setting, though this will add additional load to both {kib} and {es} instances in the cluster, as they will peform more queries.
+
+
+[float]
+[[task-manager-health-tasks-run-late]]
+=== Tasks Run Late
+
+*Symptoms*:
+
+By far the most common symptom of an underlying problem in Task manager is that tasks appear to be running late.
+For instance, recurring Tasks might run at an inconsistent cadence, or long after their scheduled time.
+
+*Resolution*:
+
+By default {kib} polls for tasks at a rate of 10 tasks every 3 seconds.
+
+This means that if many tasks have been scheduled to run at the same time, pending tasks will queue in {es}. Each {kib} instance then polls for pending tasks at a rate of up to 10 tasks at a time, at 3 second intervals. It is possible for pending tasks in the queue to exceed this capacity and run late as a result.
+
+We refer to this sort of delay as _drift_. As the root cause for _drift_ depend on the specific usage, there are no hard and fast rules to addressing them.
+
+For example, among other things, _drift_:
+
+* Is caused by *an excess of concurrent tasks* relative to the available capacity of {kib} instances in the cluster, which can be addressed by expanding the throughput of the cluster.
+* Is caused by *long running tasks* overruning their scheduled cadence, which can be addressed by reconfiguring the tasks in question.
+
+Following the <<task-manager-diagnosing-root-cause>> step-by-step guide should aid in identifying the correct resolution.
+
+_Drift_ is often be addressed by adjusting the scaling the deployment to better suit your usage.
+For details on scaling Task Manager, see <<task-manager-scaling-guidance>>.
+
+[float]
+[[task-manager-diagnosing-root-cause]]
+==== Diagnosing A Root Cause For Drift
+
+The following is a step-by-step guide to identifying a root cause for _drift_ by making sense of the output from the <<task-manager-health-monitoring>> endpoint.
+
+Retrieve the latest monitored health stats of a {kib} instance Task Manager:
+
+[source,sh]
+--------------------------------------------------
+$ curl -X GET api/task_manager/_health
+--------------------------------------------------
```

Comment:
Do you think we should add a section to the API docs about this endpoint? It could just reference this section for all the details.

Reply:
Yeah, good shout 👍